From 3d17e779a5eedc0417c1eba51626758c98032f1e Mon Sep 17 00:00:00 2001
From: Ankit Gupta <agupta@cyngn.com>
Date: Fri, 19 Feb 2016 09:33:56 -0800
Subject: [PATCH 46/74] fmapp2: Allow value for the default fm recording
 duration to be overlayed.

Issue-id: YAM-144
Change-Id: Iea23ae9842012aab35d2291c75edf406af921bbd
(cherry picked from commit 0ac486a41abf2eca3cd0412465dc6273bcdb7872)
---
 fmapp2/res/values/integers.xml                      |  1 +
 fmapp2/src/com/caf/fmradio/FmSharedPreferences.java | 11 ++++++++++-
 fmapp2/src/com/caf/fmradio/Settings.java            |  2 +-
 3 files changed, 12 insertions(+), 2 deletions(-)

diff --git a/fmapp2/res/values/integers.xml b/fmapp2/res/values/integers.xml
index a1698e1..dd17e2d 100644
--- a/fmapp2/res/values/integers.xml
+++ b/fmapp2/res/values/integers.xml
@@ -30,4 +30,5 @@
 <resources xmlns:xliff="urn:oasis:names:tc:xliff:document:1.2">
     <!-- Look at FmSharedPreferences.java for index -->
     <integer name="default_country_index">0</integer>
+    <integer name="default_record_duration_index">0</integer>
 </resources>
diff --git a/fmapp2/src/com/caf/fmradio/FmSharedPreferences.java b/fmapp2/src/com/caf/fmradio/FmSharedPreferences.java
index 5e1465a..43e1898 100644
--- a/fmapp2/src/com/caf/fmradio/FmSharedPreferences.java
+++ b/fmapp2/src/com/caf/fmradio/FmSharedPreferences.java
@@ -173,6 +173,7 @@ public class FmSharedPreferences
    private static int mLastAudioMode = -1;
 
    public static int mDefaultCountryIndex = REGIONAL_BAND_NORTH_AMERICA;
+   public static int mDefaultDurationIndex = 0;
 
    FmSharedPreferences(Context context){
       mContext = context.getApplicationContext();
@@ -455,7 +456,15 @@ public class FmSharedPreferences
       }
       SharedPreferences sp = mContext.getSharedPreferences(SHARED_PREFS, Context.MODE_PRIVATE);
       mTunedFrequency = sp.getInt(PREF_LAST_TUNED_FREQUENCY, DEFAULT_NO_FREQUENCY);
-      mRecordDuration = sp.getInt(LAST_RECORD_DURATION, RECORD_DUR_INDEX_0_VAL);
+
+      mDefaultDurationIndex = mContext.getResources().getInteger(R.integer.default_record_duration_index);
+      setRecordDuration(mDefaultDurationIndex);
+      /*
+       * Default duration on load would be either last saved duration or duration
+       * set by reading default index from integer.xml, if former is not set
+       */
+      mRecordDuration = sp.getInt(LAST_RECORD_DURATION, mRecordDuration);
+
       mAFAutoSwitch = sp.getBoolean(LAST_AF_JUMP_VALUE, true);
       mAudioOutputMode = sp.getBoolean(AUDIO_OUTPUT_MODE, true);
 
diff --git a/fmapp2/src/com/caf/fmradio/Settings.java b/fmapp2/src/com/caf/fmradio/Settings.java
index 55410fc..15aaecb 100644
--- a/fmapp2/src/com/caf/fmradio/Settings.java
+++ b/fmapp2/src/com/caf/fmradio/Settings.java
@@ -493,7 +493,7 @@ public class Settings extends PreferenceActivity implements
              if (mRxMode) {
                 mAudioPreference.setValueIndex(0);
                 if (FMRadio.RECORDING_ENABLE) {
-                    mRecordDurPreference.setValueIndex(0);
+                    mRecordDurPreference.setValueIndex(FmSharedPreferences.mDefaultDurationIndex);
                 }
                 mAfPref.setChecked(true);
                 FmSharedPreferences.SetDefaults();
-- 
2.17.1

